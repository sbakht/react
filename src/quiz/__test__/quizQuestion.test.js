import React from 'react';
import { shallow, mount, render } from 'enzyme';
import QuizQuestion from '../QuizQuestion';
import toJSON from 'enzyme-to-json';

const choices = [
  { val: 'a', text: 'riven', isWrong: true },
  { val: 'b', text: 'zed', isWrong: true },
  { val: 'c', text: 'kogmaw', isCorrect: true },
  { val: 'd', text: 'janna', isWrong: true },
];

test('render before answer', () => {
  const question = { text: 'Question title', choices };
  const fn = jest.fn();
  const component = shallow(<QuizQuestion question={question} answer={{}} onAnswer={fn} />);
  expect(toJSON(component)).toMatchSnapshot();
});

test('render after answer', () => {
  const question = { text: 'Question title', choices };
  const answer = { val: 'a' };
  const fn = jest.fn();
  const component = shallow(<QuizQuestion question={question} answer={answer} onAnswer={fn} />);
  expect(toJSON(component)).toMatchSnapshot();
});

test('can select answer', () => {
  const question = { text: 'Question title', choices };
  const fn = jest.fn();
  const component = shallow(<QuizQuestion question={question} answer={{}} onAnswer={fn} />);
  component.find('Choice').at(0).simulate('click');
  expect(fn.mock.calls.length).toBe(1);
});

test('no answer do overs', () => {
  const question = { text: 'Question title', choices };
  const answer = { val: 'a' };
  const fn = jest.fn();
  const component = shallow(<QuizQuestion question={question} answer={answer} onAnswer={fn} />);
  component.find('Choice').at(1).simulate('click');
  expect(fn.mock.calls.length).toBe(0);
});

test('allow answer do overs', () => {
  const question = { text: 'Question title', choices };
  const answer = { val: 'a' };
  const fn = jest.fn();
  const component = shallow(<QuizQuestion question={question} answer={answer} onAnswer={fn} allowDoOver />);
  component.find('Choice').at(1).simulate('click');
  expect(fn.mock.calls.length).toBe(1);
});

test('has proptypes', () => {
  expect(QuizQuestion.propTypes).toBeDefined();

  Object.keys(QuizQuestion.propTypes).map((key) => {
    expect(QuizQuestion.propTypes[key]).toBeDefined();
  });
});
