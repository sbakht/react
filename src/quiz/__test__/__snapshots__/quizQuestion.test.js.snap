// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render after answer 1`] = `
<div>
  <h3
    className="title"
  >
    Question title
  </h3>
  <Choice
    choice={
      Object {
        "isWrong": true,
        "text": "riven",
        "val": "a",
      }
    }
    isSelected={true}
    ith={1}
    onClick={[Function]}
  />
  <Choice
    choice={
      Object {
        "isWrong": true,
        "text": "zed",
        "val": "b",
      }
    }
    isSelected={false}
    ith={2}
    onClick={[Function]}
  />
  <Choice
    choice={
      Object {
        "isCorrect": true,
        "text": "kogmaw",
        "val": "c",
      }
    }
    isSelected={false}
    ith={3}
    onClick={[Function]}
  />
  <Choice
    choice={
      Object {
        "isWrong": true,
        "text": "janna",
        "val": "d",
      }
    }
    isSelected={false}
    ith={4}
    onClick={[Function]}
  />
  <AnswerMessage
    answer={
      Object {
        "val": "a",
      }
    }
  />
</div>
`;

exports[`render before answer 1`] = `
<div>
  <h3
    className="title"
  >
    Question title
  </h3>
  <Choice
    choice={
      Object {
        "isWrong": true,
        "text": "riven",
        "val": "a",
      }
    }
    isSelected={false}
    ith={1}
    onClick={[Function]}
  />
  <Choice
    choice={
      Object {
        "isWrong": true,
        "text": "zed",
        "val": "b",
      }
    }
    isSelected={false}
    ith={2}
    onClick={[Function]}
  />
  <Choice
    choice={
      Object {
        "isCorrect": true,
        "text": "kogmaw",
        "val": "c",
      }
    }
    isSelected={false}
    ith={3}
    onClick={[Function]}
  />
  <Choice
    choice={
      Object {
        "isWrong": true,
        "text": "janna",
        "val": "d",
      }
    }
    isSelected={false}
    ith={4}
    onClick={[Function]}
  />
  <AnswerMessage
    answer={Object {}}
  />
</div>
`;
